src/main/java/com/io/image/manager/service/ImageServiceImpl.java

```diff
@@ -83,7 +85,21 @@ public ImageServiceImpl(AppConfigurationProperties props, ImageCache cache, Prom
         try {
             // FIXME: this does not look pretty
             URL url = new URL(origin.getUrl() + filename);
-            byte[] imgBytes = url.openStream().readAllBytes();
+            URLConnection conn = url.openConnection();
+            Map<String,List<String>> headers = conn.getHeaderFields();
+            Optional<List<String>> cacheHead;
```

Comment:
You can start with `cacheHead = Optional.empty()`.

Do you have any idea how we can then store this information about cache? I just think that retrieving it for nothing does not make much sense right now.

Reply:
I have foucsed about getting needed cache inforamtion but have not considered how it may be used effectively

Reply:
also thanks for tha suggestion about empty optional i have no idea why i wrote it the way i did